### ISSUE TITLE
[UI] Convert `Navigator.js` from class based component to functional component

### ISSUE THREAD

**Issue by Rajdip019** [meshery/meshery#8798]
### Current Behavior
`Navigator.js` is a class-based component.

### Desired Behavior
`Navigator.js` should be a functional component.

### Implementation
Convert all the Class-based components to new updated functional components. It is important here to convert a class-based component to a functional component you may need to convert some functions as well. Check if the function is used in other components or not.

Also, make sure to use the hooks while converting them. For example, you need to covert withNotify with useNotification, useRouter hook. Please check if a hook for functional components is available and use that.

Here is a quick blog that will help you get started with : [Link](https://www.digitalocean.com/community/tutorials/five-ways-to-convert-react-class-components-to-functional-components-with-react-hooks)

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

**Comment by Rajdip019** [comment]
@SoNiC-HeRE comment on this to get assigned.

**Comment by Arjunkrishna17** [comment]
@Rajdip019 I would like to take up.

**Comment by SoNiC-HeRE** [comment]
@Rajdip019 please assign me the ```Navigator.Js``` component

**Comment by kundan766** [comment]
I want to work on this issue. please assign me this issue

**Comment by Rajdip019** [comment]
Hey @SoNiC-HeRE any update on this one?

**Comment by SoNiC-HeRE** [comment]
> Hey @SoNiC-HeRE any update on this one? 

Yeah, I'm working on this; the file is relatively large and has more components to be refactored so it's taking me time; I'll get this done asap

**Comment by Rajdip019** [comment]
Sure no problem. Do things step by step by making PR and giving updates. That is a better way to do it.

**Comment by SoNiC-HeRE** [comment]
> Sure no problem. Do things step by step by making PR and giving updates. That is a better way to do it. 

Hi, I want to regretfully inform that my pc ran into some problems ( OS crash ) and it would  take me longer for this issue to be resolved. You can unassign me and assign someone else if that's urgent.

**Comment by kundan766** [comment]
Yes ,sure you can assign me the issue

**Comment by EraKin575** [comment]
@kundan766  any updates?

**Comment by kundan766** [comment]
working on it .soon i will update  you

**Comment by MauryanKansara** [comment]
Hey @Rajdip019 can you assign this issue to me? The assigned contributor does not seem to be active.

**Comment by Aseem22** [comment]
Can you assign this task to me, please? @Rajdip019 @saurabh100ni @EraKin575

**Comment by saurabh100ni** [comment]
Go on 😎🚀 @Aseem22

**Comment by akshansh-modi** [comment]
is this issue still open , i am interested on working on this issue please assign this to me

**Comment by ishwar170695** [comment]
The requested migration has already been completed. `Navigator.js` no longer exists and has been replaced by `ui/components/layout/Navigator/Navigator.tsx`, which exports a functional component (`NavigatorWrapper`/`NavigatorContent`) implemented with React hooks (`useState`, `useEffect`, `useMemo`, `useSelector`, `useDispatch`, `useRouter`). No class-based Navigator implementation remains.
